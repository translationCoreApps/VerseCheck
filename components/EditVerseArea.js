import React from 'react'
import {Checkbox, Glyphicon, FormGroup, FormControl} from 'react-bootstrap'
import style from '../css/Style';

let EditVerseArea = (props) => {
  const tagList1 = [
    ["spelling", "Spelling"],
    ["punctuation", "Punctuation"],
    ["wordChoice", "Word Choice"],
  ]

  const tagList2 = [
    ["meaning", "Meaning"],
    ["grammar", "Grammar"],
    ["other", "Other"]
  ]

  const checkboxesColumn1 = tagList1.map(tag =>
    <Checkbox
      key={tag[0]}
      checked={props.tags.includes(tag[0])}
      disabled={!props.verseChanged}
      style={props.verseChanged ? {width: '33.3%', marginLeft: '10px', marginRight: '10px', color: 'var(--text-color-dark)'} : {width: '33.3%', marginLeft: '10px', marginRight: '10px', color: 'var(--text-color-light)'}}
      onChange={props.actions.handleTagsCheckbox.bind(this, tag[0])}
    >
      {tag[1]}
    </Checkbox>
  )

  const checkboxesColumn2 = tagList2.map(tag =>
    <Checkbox
      key={tag[0]}
      checked={props.tags.includes(tag[0])}
      disabled={!props.verseChanged}
      style={props.verseChanged ? {width: '33.3%', marginLeft: '10px', marginRight: '10px', color: 'var(--text-color-dark)'} : {width: '33.3%', marginLeft: '10px', marginRight: '10px', color: 'var(--text-color-light)'}}
      onChange={props.actions.handleTagsCheckbox.bind(this, tag[0])}
    >
      {tag[1]}
    </Checkbox>
  )

  let checkBoxText = props.verseChanged ? "Next, select reason(s) for change" : "First, make changes to verse above";

  return (
    <div style={style.editArea}>
      <div style={{fontWeight: 'bold'}}>
        <Glyphicon glyph='pencil' style={{marginRight: '5px'}} />
        Edit Verse
      </div>
      <FormGroup style={{flex: 'auto', display: 'flex', flexDirection: 'column', marginBottom: '5px'}} controlId='formControlsTextarea'>
        <FormControl autoFocus
          componentClass='textarea'
          type='text'
          defaultValue={props.verseText}
          style={{flex: 'auto', direction: props.dir}}
          onBlur={props.actions.handleEditVerse.bind(this)}
          onInput={props.actions.checkVerse.bind(this)}
        />
      <div style={{flex: '0 0 65px', marginTop: '5px', fontSize: '0.9em'}}>
        {checkBoxText}
        <br />
        <div style={{ display: 'flex' }}>
          <div style={{ flex: '1' }}>
            {checkboxesColumn1}
          </div>
          <div style={{ flex: '1' }}>
            {checkboxesColumn2}
          </div>
        </div>
      </div>
      </FormGroup>
    </div>
  )
}

export default EditVerseArea;
