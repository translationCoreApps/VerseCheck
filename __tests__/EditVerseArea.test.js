/* eslint-env jest */
import React from 'react';
import EditVerseArea from '../src/components/EditVerseArea';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('EditVerseArea component Tests', () => {
  let props;
  const mock_translate = (text) => { return text; };

  beforeEach(()=> {
    props = {
      translate: mock_translate,
      tags: [],
      verseChanged: false,
      verseText: 'This is the verse text',
      dir: 'ltr',
      actions: {
        handleTagsCheckbox: () => jest.fn(),
        handleEditVerse: () => jest.fn(),
        checkVerse: () => jest.fn(),
      }
    };
  });

  test('Check EditVerseArea component', () => {
    const wrapper = shallow(<EditVerseArea {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.text()).toEqual('<Glyphicon />edit_verse<FormGroup />');
  });

  test('Check EditVerseArea component select all', () => {
    props.tags = ["spelling", "punctuation", "wordChoice", "meaning", "grammar", "other"];
    const wrapper = shallow(<EditVerseArea {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.text()).toEqual('<Glyphicon />edit_verse<FormGroup />');
  });

  test('Check EditVerseArea component select all', () => {
    props.verseChanged = true;
    const wrapper = shallow(<EditVerseArea {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.text()).toEqual('<Glyphicon />edit_verse<FormGroup />');
  });

});
