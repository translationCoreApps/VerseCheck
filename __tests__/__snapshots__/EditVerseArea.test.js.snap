// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditVerseArea component Tests Check EditVerseArea component 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flex": "auto",
      "flexDirection": "column",
      "height": "100%",
      "padding": "5px 15px 0 15px",
    }
  }
>
  <div
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="pencil"
      style={
        Object {
          "marginRight": "5px",
        }
      }
    />
    edit_verse
  </div>
  <FormGroup
    bsClass="form-group"
    controlId="formControlsTextarea"
    style={
      Object {
        "display": "flex",
        "flex": "auto",
        "flexDirection": "column",
        "marginBottom": "5px",
      }
    }
  >
    <FormControl
      autoFocus={true}
      bsClass="form-control"
      componentClass="textarea"
      defaultValue="This is the verse text"
      onBlur={[Function]}
      onInput={[Function]}
      style={
        Object {
          "direction": "ltr",
          "flex": "auto",
        }
      }
      type="text"
    />
    <div
      style={
        Object {
          "flex": "0 0 65px",
          "fontSize": "0.9em",
          "marginTop": "5px",
        }
      }
    >
      First, make changes to verse above
      <br />
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <div
          style={
            Object {
              "flex": "1",
            }
          }
        >
          <Checkbox
            bsClass="checkbox"
            checked={false}
            disabled={true}
            inline={false}
            key="spelling"
            onChange={[Function]}
            style={
              Object {
                "color": "var(--text-color-light)",
                "marginLeft": "10px",
                "marginRight": "10px",
              }
            }
            title=""
          >
            spelling
          </Checkbox>
          <Checkbox
            bsClass="checkbox"
            checked={false}
            disabled={true}
            inline={false}
            key="punctuation"
            onChange={[Function]}
            style={
              Object {
                "color": "var(--text-color-light)",
                "marginLeft": "10px",
                "marginRight": "10px",
              }
            }
            title=""
          >
            punctuation
          </Checkbox>
          <Checkbox
            bsClass="checkbox"
            checked={false}
            disabled={true}
            inline={false}
            key="wordChoice"
            onChange={[Function]}
            style={
              Object {
                "color": "var(--text-color-light)",
                "marginLeft": "10px",
                "marginRight": "10px",
              }
            }
            title=""
          >
            word_choice
          </Checkbox>
        </div>
        <div
          style={
            Object {
              "flex": "1",
            }
          }
        >
          <Checkbox
            bsClass="checkbox"
            checked={false}
            disabled={true}
            inline={false}
            key="meaning"
            onChange={[Function]}
            style={
              Object {
                "color": "var(--text-color-light)",
                "marginLeft": "10px",
                "marginRight": "10px",
              }
            }
            title=""
          >
            meaning
          </Checkbox>
          <Checkbox
            bsClass="checkbox"
            checked={false}
            disabled={true}
            inline={false}
            key="grammar"
            onChange={[Function]}
            style={
              Object {
                "color": "var(--text-color-light)",
                "marginLeft": "10px",
                "marginRight": "10px",
              }
            }
            title=""
          >
            grammar
          </Checkbox>
          <Checkbox
            bsClass="checkbox"
            checked={false}
            disabled={true}
            inline={false}
            key="other"
            onChange={[Function]}
            style={
              Object {
                "color": "var(--text-color-light)",
                "marginLeft": "10px",
                "marginRight": "10px",
              }
            }
            title=""
          >
            other
          </Checkbox>
        </div>
      </div>
    </div>
  </FormGroup>
</div>
`;

exports[`EditVerseArea component Tests Check EditVerseArea component select all 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flex": "auto",
      "flexDirection": "column",
      "height": "100%",
      "padding": "5px 15px 0 15px",
    }
  }
>
  <div
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="pencil"
      style={
        Object {
          "marginRight": "5px",
        }
      }
    />
    edit_verse
  </div>
  <FormGroup
    bsClass="form-group"
    controlId="formControlsTextarea"
    style={
      Object {
        "display": "flex",
        "flex": "auto",
        "flexDirection": "column",
        "marginBottom": "5px",
      }
    }
  >
    <FormControl
      autoFocus={true}
      bsClass="form-control"
      componentClass="textarea"
      defaultValue="This is the verse text"
      onBlur={[Function]}
      onInput={[Function]}
      style={
        Object {
          "direction": "ltr",
          "flex": "auto",
        }
      }
      type="text"
    />
    <div
      style={
        Object {
          "flex": "0 0 65px",
          "fontSize": "0.9em",
          "marginTop": "5px",
        }
      }
    >
      First, make changes to verse above
      <br />
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <div
          style={
            Object {
              "flex": "1",
            }
          }
        >
          <Checkbox
            bsClass="checkbox"
            checked={true}
            disabled={true}
            inline={false}
            key="spelling"
            onChange={[Function]}
            style={
              Object {
                "color": "var(--text-color-light)",
                "marginLeft": "10px",
                "marginRight": "10px",
              }
            }
            title=""
          >
            spelling
          </Checkbox>
          <Checkbox
            bsClass="checkbox"
            checked={true}
            disabled={true}
            inline={false}
            key="punctuation"
            onChange={[Function]}
            style={
              Object {
                "color": "var(--text-color-light)",
                "marginLeft": "10px",
                "marginRight": "10px",
              }
            }
            title=""
          >
            punctuation
          </Checkbox>
          <Checkbox
            bsClass="checkbox"
            checked={true}
            disabled={true}
            inline={false}
            key="wordChoice"
            onChange={[Function]}
            style={
              Object {
                "color": "var(--text-color-light)",
                "marginLeft": "10px",
                "marginRight": "10px",
              }
            }
            title=""
          >
            word_choice
          </Checkbox>
        </div>
        <div
          style={
            Object {
              "flex": "1",
            }
          }
        >
          <Checkbox
            bsClass="checkbox"
            checked={true}
            disabled={true}
            inline={false}
            key="meaning"
            onChange={[Function]}
            style={
              Object {
                "color": "var(--text-color-light)",
                "marginLeft": "10px",
                "marginRight": "10px",
              }
            }
            title=""
          >
            meaning
          </Checkbox>
          <Checkbox
            bsClass="checkbox"
            checked={true}
            disabled={true}
            inline={false}
            key="grammar"
            onChange={[Function]}
            style={
              Object {
                "color": "var(--text-color-light)",
                "marginLeft": "10px",
                "marginRight": "10px",
              }
            }
            title=""
          >
            grammar
          </Checkbox>
          <Checkbox
            bsClass="checkbox"
            checked={true}
            disabled={true}
            inline={false}
            key="other"
            onChange={[Function]}
            style={
              Object {
                "color": "var(--text-color-light)",
                "marginLeft": "10px",
                "marginRight": "10px",
              }
            }
            title=""
          >
            other
          </Checkbox>
        </div>
      </div>
    </div>
  </FormGroup>
</div>
`;

exports[`EditVerseArea component Tests Check EditVerseArea component select all 2`] = `
<div
  style={
    Object {
      "display": "flex",
      "flex": "auto",
      "flexDirection": "column",
      "height": "100%",
      "padding": "5px 15px 0 15px",
    }
  }
>
  <div
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="pencil"
      style={
        Object {
          "marginRight": "5px",
        }
      }
    />
    edit_verse
  </div>
  <FormGroup
    bsClass="form-group"
    controlId="formControlsTextarea"
    style={
      Object {
        "display": "flex",
        "flex": "auto",
        "flexDirection": "column",
        "marginBottom": "5px",
      }
    }
  >
    <FormControl
      autoFocus={true}
      bsClass="form-control"
      componentClass="textarea"
      defaultValue="This is the verse text"
      onBlur={[Function]}
      onInput={[Function]}
      style={
        Object {
          "direction": "ltr",
          "flex": "auto",
        }
      }
      type="text"
    />
    <div
      style={
        Object {
          "flex": "0 0 65px",
          "fontSize": "0.9em",
          "marginTop": "5px",
        }
      }
    >
      Next, select reason(s) for change
      <br />
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <div
          style={
            Object {
              "flex": "1",
            }
          }
        >
          <Checkbox
            bsClass="checkbox"
            checked={false}
            disabled={false}
            inline={false}
            key="spelling"
            onChange={[Function]}
            style={
              Object {
                "color": "var(--text-color-dark)",
                "marginLeft": "10px",
                "marginRight": "10px",
              }
            }
            title=""
          >
            spelling
          </Checkbox>
          <Checkbox
            bsClass="checkbox"
            checked={false}
            disabled={false}
            inline={false}
            key="punctuation"
            onChange={[Function]}
            style={
              Object {
                "color": "var(--text-color-dark)",
                "marginLeft": "10px",
                "marginRight": "10px",
              }
            }
            title=""
          >
            punctuation
          </Checkbox>
          <Checkbox
            bsClass="checkbox"
            checked={false}
            disabled={false}
            inline={false}
            key="wordChoice"
            onChange={[Function]}
            style={
              Object {
                "color": "var(--text-color-dark)",
                "marginLeft": "10px",
                "marginRight": "10px",
              }
            }
            title=""
          >
            word_choice
          </Checkbox>
        </div>
        <div
          style={
            Object {
              "flex": "1",
            }
          }
        >
          <Checkbox
            bsClass="checkbox"
            checked={false}
            disabled={false}
            inline={false}
            key="meaning"
            onChange={[Function]}
            style={
              Object {
                "color": "var(--text-color-dark)",
                "marginLeft": "10px",
                "marginRight": "10px",
              }
            }
            title=""
          >
            meaning
          </Checkbox>
          <Checkbox
            bsClass="checkbox"
            checked={false}
            disabled={false}
            inline={false}
            key="grammar"
            onChange={[Function]}
            style={
              Object {
                "color": "var(--text-color-dark)",
                "marginLeft": "10px",
                "marginRight": "10px",
              }
            }
            title=""
          >
            grammar
          </Checkbox>
          <Checkbox
            bsClass="checkbox"
            checked={false}
            disabled={false}
            inline={false}
            key="other"
            onChange={[Function]}
            style={
              Object {
                "color": "var(--text-color-dark)",
                "marginLeft": "10px",
                "marginRight": "10px",
              }
            }
            title=""
          >
            other
          </Checkbox>
        </div>
      </div>
    </div>
  </FormGroup>
</div>
`;
